components/ai_chat/core/browser/conversation_driver_unittest.cc

```diff
@@ -0,0 +1,233 @@
+      base::RepeatingCallback<mojo::PendingRemote<skus::mojom::SkusService>()>
+          skus_service_getter,
+      scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+      std::string_view channel_string)
+      : ConversationDriver(profile_prefs,
+                           local_state,
+                           ai_chat_metrics,
+                           std::move(skus_service_getter),
+                           url_loader_factory,
+                           channel_string) {}
+  ~MockConversationDriver() override = default;
+
+  MOCK_METHOD(GURL, GetPageURL, (), (const, override));
+  MOCK_METHOD(std::u16string, GetPageTitle, (), (const, override));
+  MOCK_METHOD(void,
+              GetPageContent,
+              (GetPageContentCallback, std::string_view),
+              (override));
+};
+
+}  // namespace
+
+class ConversationDriverUnitTest : public testing::Test {
+ public:
+  ConversationDriverUnitTest() = default;
+  ~ConversationDriverUnitTest() override = default;
+
+  void SetUp() override {
+    prefs::RegisterProfilePrefs(prefs_.registry());
+    prefs::RegisterLocalStatePrefs(local_state_.registry());
+    shared_url_loader_factory_ =
+        base::MakeRefCounted<network::WeakWrapperSharedURLLoaderFactory>(
+            &url_loader_factory_);
+    auto skus_service_getter = base::BindRepeating(
+        []() { return mojo::PendingRemote<skus::mojom::SkusService>(); });
+    // set prefs::kDefaultModelKey to llama or claude first, then change model
+    // later
+    conversation_driver_ = std::make_unique<MockConversationDriver>(
+        &prefs_, &local_state_, nullptr, skus_service_getter,
+        shared_url_loader_factory_, "");
+  }
+
+  void TearDown() override {}
+
+ protected:
+  base::test::TaskEnvironment task_environment_;
+  sync_preferences::TestingPrefServiceSyncable prefs_;
+  sync_preferences::TestingPrefServiceSyncable local_state_;
+  network::TestURLLoaderFactory url_loader_factory_;
+  scoped_refptr<network::SharedURLLoaderFactory> shared_url_loader_factory_;
+  data_decoder::test::InProcessDataDecoder in_process_data_decoder_;
+  std::unique_ptr<MockConversationDriver> conversation_driver_;
+};
+
+TEST_F(ConversationDriverUnitTest, SummarizeSelectedText) {
+  bool is_page_content_linked = false;
+  url_loader_factory_.SetInterceptor(
+      base::BindLambdaForTesting([&](const network::ResourceRequest& request) {
+        std::string body = network::GetUploadData(request);
+        EXPECT_NE(body.find("I have spoken."), std::string::npos);
+        EXPECT_NE(body.find(l10n_util::GetStringUTF8(
+                      IDS_AI_CHAT_QUESTION_SUMMARIZE_SELECTED_TEXT)),
+                  std::string::npos);
+        if (is_page_content_linked) {
+          EXPECT_NE(body.find("The child's name is Grogu."), std::string::npos);
+        }
+        // Set header for enabling SSE.
+        auto head = network::mojom::URLResponseHead::New();
+        head->mime_type = "text/event-stream";
```

Comment:
by the way, we can set a featureparam to force SSE disabled, but this seems simple enough for now

Reply:
Ideally we should test both cases in the end, I choose this as a starter as that's what my env was using.
Comprehensive unit test coverage of ConversationDriver is out of the PR scope, so here is just a starter for adding tests.